python/tests/test_highlevel.py

```diff
@@ -4261,3 +4261,24 @@ def test_macs_error(self):
             ValueError, match="macs output only supports single letter alleles"
         ):
             ts.to_macs()
+
+
+class TestTreeSequenceGetSite:
```

Comment:
The tests at test_highlevel.py:2006 are specifically for the site method. These tests are a good start, but I think adding 
```python
s3 = ts.site(position=site.postion)
assert s3 = site
```
would be an good thing to add too.